crates/core/tedge_api/src/workflow/state.rs

```diff
@@ -258,21 +258,44 @@ impl GenericCommandState {
     /// - The script command is first tokenized using shell escaping rules.
     ///   `/some/script.sh arg1 "arg 2" "arg 3"` -> ["/some/script.sh", "arg1", "arg 2", "arg 3"]
     /// - Then each token matching `${x.y.z}` is substituted with the value pointed by the JSON path.
-    pub fn inject_parameters(&self, args: &[String]) -> Vec<String> {
-        args.iter().map(|arg| self.inject_parameter(arg)).collect()
+    pub fn inject_values_into_parameters(&self, args: &[String]) -> Vec<String> {
+        args.iter()
+            .map(|arg| self.inject_values_into_template(arg))
+            .collect()
     }
 
-    /// Inject values extracted from the message payload into a script argument
+    /// Inject values extracted from the message payload into a template string
+    ///
+    /// - Search the template string for path patterns `${...}`
+    /// - Replace all these paths by the value extracted from self using the paths
+    ///
+    /// `"prefix-${.payload.x}-separator-${.payload.y}-suffix"` is replaced by
+    /// `"prefix-X-separator-Y-suffix"` in a context where the payload is `{"x":"X", "y":"Y"}`
+    pub fn inject_values_into_template(&self, script_parameter: &str) -> String {
```

Comment:
Minor(since this function is more generic now):
```suggestion
    pub fn inject_values_into_template(&self, target: &str) -> String {
```

Reply:
Renamed